### ISSUE TITLE
"location" paramter should work if it does not exist in azure.conf

### ISSUE THREAD

**Issue by weherdh** [kubernetes/kubernetes#61265]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug

**What happened**:
Failed to provision pv with storage class if there is no location in azure.conf
# cat /etc/azure.conf
{
  "aadClientID" : "xxxxxxxxxxxxxxxxxxx",
  "aadClientSecret" : "xxxxxxxxxxxxxxxxxx",
  "subscriptionID" : "xxxxxxxxxxxxxxxxxxxxxx",
  "tenantID" : "xxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "resourceGroup": "openshift-qe",
}

# kubectl describe sc
Name:            sc-x2cij
IsDefaultClass:  No
Annotations:     <none>
Provisioner:     kubernetes.io/azure-disk
Parameters:      location=eastus,skuName=Standard_RAGRS
ReclaimPolicy:   Delete
Events:          <none>

# kubectl get pvc
NAME      STATUS    VOLUME    CAPACITY   ACCESS MODES   STORAGECLASS   AGE
azpvc     Pending                                       sc-x2cij       14m

# oc describe pvc azpvc
Name:          azpvc
Namespace:     x2cij
StorageClass:  sc-x2cij
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-class=sc-x2cij
               volume.beta.kubernetes.io/storage-provisioner=kubernetes.io/azure-disk
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
Events:
  Type     Reason              Age                From                         Message
  ----     ------              ----               ----                         -------
  Warning  ProvisioningFailed  4m (x43 over 14m)  persistentvolume-controller  Failed to provision volume with StorageClass "sc-x2cij": Create Storage Account: pvc3784877905001, error: storage.AccountsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="LocationRequired" Message="The location property is required for this definition."

**What you expected to happen**:
Because storage class has parameter of "location", so the location in azure.conf should be optional. Otherwise, what is the "location" parameter used for?

**How to reproduce it (as minimally and precisely as possible)**:
As "What happened"

**Anything else we need to know?**:

**Environment**: 
- Kubernetes version (use `kubectl version`):
# kubectl version
Client Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.1+a0ce1bc657", GitCommit:"a0ce1bc", GitTreeState:"clean", BuildDate:"2018-03-06T16:19:01Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.1+a0ce1bc657", GitCommit:"a0ce1bc", GitTreeState:"clean", BuildDate:"2018-03-06T16:19:01Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
RHEL 7.4
- Kernel (e.g. `uname -a`):
- Install tools:
ansible
- Others:

**Comment by weherdh** [comment]
cc @rootfs

**Comment by andyzhangx** [comment]
@weherdh You are using default kind `Shared` which means the shared storage accounts will be created before any pvc, if you specify `Dedicated` kind, it will respect `location` value in storage class.

**Comment by weherdh** [comment]
Yes, after I added `kind: Dedicated` parameter, it works. So the location in azure.conf should still be optional.

**Comment by andyzhangx** [comment]
@weherdh I will add a note in doc that kind `Shared` don't respect `location`, `storageaccount` parameter
